src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,184 @@
+            RobotMap.Solenoids.LiftTransmission2);
+
+    public enum LiftHeight {
+        //will be changed
+        kFloor(0), kSwitch(1), kPortal(2), kIntake(3), kScaleLow(4), kScaleHigh(5), kClimb(6);
+
+        @SuppressWarnings("MemberName")
+        private double value;
+
+        LiftHeight(double value) {
+            this.value = value;
+        }
+
+        public double get() {
+            return value;
+        }
+    }
+
+    public Lift() {
+        addChild(liftEncoder);
+        //will be changed
+        liftEncoder.setDistancePerPulse(.1);
+    }
+
+    private Boolean TopLimitSwitchOn() {
+        if (TopLimitSwitchLift.get())
+            return true;
+        else
+            return false;
+    }
+
+    private Boolean BottomLimitSwitchOn() {
+        if (BottomLimitSwitchLift.get())
+            return true;
+        else
+            return false;
+    }
+
+    //gets the Lift Encoder's distance
+    public double EncoderDistance = liftEncoder.getDistance();
+    //gets the Lift Encoder's speed
+    public double EncoderSpeed = liftEncoder.getRate();
+
+    //sets the motor's speed, can be called later
+    public void goUp() {
+        liftdrive.set(0.25);
+    }
+
+    public void goDown() {
+        liftdrive.set(-0.25);
+    }
+
+    //gear changes
+    public void shiftToHighGear() {
+        if (LiftTransmission.get() != Value.kForward) {
+            LiftTransmission.set(Value.kForward);
+        }
+    }
+
+    public void shiftToLowGear() {
+        if (LiftTransmission.get() != Value.kReverse) {
+            LiftTransmission.set(Value.kReverse);
+        }
+    }
+
+    public void initDefaultCommand() {
+        setDefaultCommand(new SubsystemCommand(this) {
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        });
+    }
+
+    public Command GoToHeight(LiftHeight height) {
+        return new SubsystemCommand(this) {
+            @Override
+            protected void initialize() {
+                shiftToHighGear();
+                if (height.value > EncoderDistance) {
+                    goUp();
+                } else
+                    goDown();
+            }
+
+            @Override
+            protected void execute() {
+
+            }
+
+            @Override
+            protected boolean isFinished() {
+                if (EncoderDistance == height.value) {
+                    return true;
+                } else
+                    return false;
+            }
+
+            @Override
+            protected void end() {
+                liftdrive.stopMotor();
+            }
+        };
+    }
+
+    public Command ClimbUp() {
+        return new CommandChain("Climb Up").then(ShiftToHighGear(), GoToHeight(LiftHeight.kClimb), ShiftToLowGear(),
+                GoToHeight(LiftHeight.kScaleLow));
+    }
+
+    public Command GoUp() {
+        return new ActionCommand("Go Up", this, this::goUp);
```

Comment:
This should not be an action command, it should only go up until the top limit switch is pressed.